src/components/Similarity/SimilarFilesSidebar.tsx

```diff
@@ -83,17 +86,42 @@ const SimilarEntriesComponent: React.FC<SimilarEntriesComponentProps> = ({
     };
   }, [filePath]);
 
+  const updateSimilarEntries = async () => {
+    const searchResults = await performSearch(filePath);
+    if (searchResults.length > 0) {
+      setSimilarEntries(searchResults);
+    }
+  };
+
   return (
-    <div className="h-full overflow-y-auto overflow-x-hidden mt-0 border-l-[0.1px] border-t-0 border-b-0 border-r-0 border-gray-600 border-solid">
-      {similarEntries.length > 0 && (
-        <div className="flex items-center justify-center bg-gray-800 mt-0 mb-0 p-0">
-          <PiGraph className="text-white mt-1" />
+    <div
+      className={`h-below-titlebar ${
+        similarEntries.length > 0 ? "overflow-y-auto" : "overflow-y-hidden"
+      } overflow-x-hidden mt-0 border-l-[0.1px] border-t-0 border-b-0 border-r-0 border-gray-600 border-solid`}
+    >
+      {" "}
+      {/* {similarEntries.length > 0 && ( */}
+      <div className="flex items-center bg-gray-800 p-0">
+        {/* Invisible Spacer */}
+        <div className="flex-1"></div>
 
-          <p className="text-gray-200 text-sm pl-1 mb-0  mt-1 pt-0 pb-0">
-            Related Notes
-          </p>
+        {/* Centered content: PiGraph icon and Related Notes text */}
+        <div className="flex items-center justify-center px-4">
+          <PiGraph className="text-gray-300 mt-1" />
+          <p className="text-gray-300 text-sm pl-1 mb-0 mt-1">Related Notes</p>
         </div>
-      )}
+
+        <div
+          className="flex-1 flex justify-end pr-3 pt-1 cursor-pointer"
+          onClick={() => {
+            setHitRefresh(true);
+            setSimilarEntries([]); // simulate refresh
+            updateSimilarEntries();
+          }}
```

Comment:
Perhaps it would be simpler just manage the hitRefresh state without actually touching the useEffect?

The way we could do that is at the end of the onClick listener, you can just set `hitRefresh` to be false. Just have to turn this into an async listener and wait for the update to be done before resetting the state

Reply:
hitRefresh is used only to update the message below and not to do any actual refreshing:
{!hitRefresh ? (
              <>Hit refresh to show related notes...</>
            ) : (
              <>Make sure your note is not empty...</>
            )}